From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sun, 15 May 2022 21:40:03 +0200
Subject: [PATCH] {BUG in 5.18-rc1} drivers/gpu/drm/imx/ipuv3-crtc: remove & to
 check for NULL address
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gcc 12.1.0 warns for x86-64:

    drivers/gpu/drm/imx/ipuv3-crtc.c: In function ‘ipu_crtc_disable_planes’:
    drivers/gpu/drm/imx/ipuv3-crtc.c:72:21: error: the comparison will
    always evaluate as ‘true’ for the address of ‘plane’ will never be
    NULL [-Werror=address]
       72 |                 if (&ipu_crtc->plane[1] && plane == &ipu_crtc->plane[1]->base)
          |                     ^
    drivers/gpu/drm/imx/ipuv3-crtc.c:37:34: note: ‘plane’ declared here
       37 |         struct ipu_plane        *plane[2];
          |                                  ^~~~~
---
 drivers/gpu/drm/imx/ipuv3-crtc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/imx/ipuv3-crtc.c b/drivers/gpu/drm/imx/ipuv3-crtc.c
index 9c8829f945b2..f7863d6dea80 100644
--- a/drivers/gpu/drm/imx/ipuv3-crtc.c
+++ b/drivers/gpu/drm/imx/ipuv3-crtc.c
@@ -69,7 +69,7 @@ static void ipu_crtc_disable_planes(struct ipu_crtc *ipu_crtc,
 	drm_atomic_crtc_state_for_each_plane(plane, old_crtc_state) {
 		if (plane == &ipu_crtc->plane[0]->base)
 			disable_full = true;
-		if (&ipu_crtc->plane[1] && plane == &ipu_crtc->plane[1]->base)
+		if (ipu_crtc->plane[1] && plane == &ipu_crtc->plane[1]->base)
 			disable_partial = true;
 	}
 
-- 
